terraform/Makefile

```diff
@@ -60,11 +60,28 @@ reapply:
 	terraform -chdir=$(TFDIR) apply -var-file="vars.tfvars"
 
 # Regenerates types_terraform.go
+#
+# The proto files needed for this generator exist only inside the go mod cache,
+# so we retrieve the file path for the cached proto files with go mod tools.
+gen-schema: API_MOD_PATH ?= $(shell go mod download --json github.com/gravitational/teleport/api | jq .Dir)
+gen-schema: PROTOBUF_MOD_PATH ?= $(shell go mod download --json github.com/gogo/protobuf | jq .Dir)
+# protoc expects to find .proto files needed by types.proto by its full path.
+# Due to the way go modules work, we don't have the exact path to the cached
+# module on file, so we make a temporary directory with a symlink to the cached 
+# module to get the desired file structure. This is a bandaid fix due to the
+# removal of /vendor, and should be replaced with a better solution. In the main
+# teleport repo, we ran into a similar issue with our grpc generation, and we used
+# the M flag to achieve a similar result, but I haven't found the way to do this here.
 gen-schema:
+	mkdir -p $(LOCALDIR)/vendor/github.com/gravitational/teleport/api
+	ln -s $(API_MOD_PATH)/* $(LOCALDIR)/vendor/github.com/gravitational/teleport/api
```

Comment:
@zmb3 This protoc issue is similar to the one we were working on the in main repo. I couldn't find any "nice" solution like the `M` flag supported by `gogofast_out`. I don't think it's worth spending too much time on this right now, but does any workaround come to mind? ( other than a symink :D )